traffic_ops/testing/api/v4/cachegroups_parameters_test.go

```diff
@@ -79,6 +90,124 @@ func CreateTestCacheGroupParameters(t *testing.T) {
 	testData.CacheGroupParameterRequests = append(testData.CacheGroupParameterRequests, resp.Response...)
 }
 
+func CreateTestCacheGroupParametersAlreadyExist(t *testing.T) {
+
+	resp, _, _ := TOSession.GetAllCacheGroupParameters(client.RequestOptions{})
+	cachegroupParameters := resp.Response
+	getAllcgparameters := cachegroupParameters.CacheGroupParameters
+	if len(getAllcgparameters) < 1 {
+		t.Fatalf("No existing cachegroup parameters available to test duplicate functionality")
+	}
+	parameterID := *getAllcgparameters[0].Parameter
+
+	//Get cachegroup id from cachegroup name
+	opts := client.NewRequestOptions()
+	opts.QueryParameters.Set("name", *getAllcgparameters[0].CacheGroup)
+
+	cgResponse, _, _ := TOSession.GetCacheGroups(opts)
+	if len(cgResponse.Response) < 1 {
+		t.Fatalf("No data available for GetCacheGroups by Name - Length %d", len(cgResponse.Response))
+	}
+	cacheGroupID := cgResponse.Response[0].ID
+	if parameterID != 0 && *cacheGroupID != 0 {
+		cgparameters, reqInf, err := TOSession.CreateCacheGroupParameter(*cacheGroupID, parameterID, client.RequestOptions{})
+		if reqInf.StatusCode != http.StatusBadRequest {
+			t.Errorf("Expected 400 status code, got %v", reqInf.StatusCode)
+		}
+		if err == nil {
+			t.Errorf("Expected Parameter - %d already associated with cachegroup %d - Alerts %v", parameterID, cacheGroupID, cgparameters.Alerts)
+		}
+	} else {
+		t.Errorf("Nil value found in one of the values. Cachegroup ID %d, Parameter ID %d", cacheGroupID, parameterID)
```

Comment:
This will print the pointer value of `cacheGroupID`, not the actual ID value to which it points.

Reply:
Fixed it.